respa_admin/views/resources.py

```diff
@@ -0,0 +1,127 @@
+from django.http import HttpResponseRedirect
+from django.template.response import TemplateResponse
+
+from django.views.generic import (
+    CreateView,
+    ListView,
+)
+
+from resources.models import Resource
+from respa_admin.views import forms
+
+from respa_admin.views.forms import (
+    ResourceForm,
+)
+
+
+def admin_index(request):
+    return TemplateResponse(request, 'index.html')
+
+
+def admin_office(request):
+    return TemplateResponse(request, 'page_office.html')
+
+
+def admin_form(request):
+    return TemplateResponse(request, 'page_form.html')
+
+
+class ResourceListView(ListView):
+    model = Resource
+    paginate_by = 10
+    context_object_name = 'resources'
+    template_name = 'page_resources.html'
+
+
+class SaveResourceView(CreateView):
+    """
+    View for saving new resources and updating existing resources.
+    """
+
+    http_method_names = ['get', 'post']
+    success_url = 'success'  # This is just for testing purposes.
+    model = Resource
+    form_class = ResourceForm
+    template_name = 'resources/create_resource.html'
+    extra_formsets = 1
+    button_value = 'Create'
+
+    def get(self, request, *args, **kwargs):
+        if kwargs:
+            self.object = Resource.objects.get(pk=kwargs['resource_id'])
+            self.extra_formsets = 0
+            self.button_value = 'Update'
+        else:
+            self.object = None
+
+        form_class = self.get_form_class()
+        form = self.get_form(form_class)
+
+        period_formset_with_days = forms.get_period_formset(
+            self.request,
+            extra=self.extra_formsets,
+            instance=self.object
+        )
+
+        resource_image_formset = forms.get_resource_image_formset(
+            self.request,
+            extra=self.extra_formsets,
+            instance=self.object
+        )
+
+        return self.render_to_response(
+            self.get_context_data(
+                form=form,
+                period_formset_with_days=period_formset_with_days,
+                resource_image_formset=resource_image_formset,
```

Comment:
Somehow I have the feeling that the period and resource image formset could be part of the same top level formset i.e. part of the resource form that could actually be a formset. But we can discuss this in other channels...